go/worker/compute/executor/committee/fault.go

```diff
@@ -1,186 +0,0 @@
-	consensus "github.com/oasisprotocol/oasis-core/go/consensus/api"
-	roothash "github.com/oasisprotocol/oasis-core/go/roothash/api"
-	"github.com/oasisprotocol/oasis-core/go/roothash/api/commitment"
-	runtimeRegistry "github.com/oasisprotocol/oasis-core/go/runtime/registry"
-)
-
-type faultSubmitter interface {
-	// SubmitExecutorCommit submits an executor commitment when a fault is detected.
-	SubmitExecutorCommit(ctx context.Context, commit *commitment.ExecutorCommitment) error
-}
-
-type nodeFaultSubmitter struct {
-	node *Node
-}
-
-// Implements faultSubmitter.
-func (nf *nodeFaultSubmitter) SubmitExecutorCommit(ctx context.Context, commit *commitment.ExecutorCommitment) error {
-	tx := roothash.NewExecutorCommitTx(0, nil, nf.node.commonNode.Runtime.ID(), []commitment.ExecutorCommitment{*commit})
-	return consensus.SignAndSubmitTx(ctx, nf.node.commonNode.Consensus, nf.node.commonNode.Identity.NodeSigner, tx)
-}
-
-func newNodeFaultSubmitter(node *Node) faultSubmitter {
-	return &nodeFaultSubmitter{node}
-}
-
-type faultDetector struct {
-	runtime   runtimeRegistry.Runtime
-	submitter faultSubmitter
-	commit    *commitment.ExecutorCommitment
-
-	quitCh  chan struct{}
-	eventCh chan *roothash.Event
-
-	logger *logging.Logger
-}
-
-func (d *faultDetector) notify(ev *roothash.Event) {
-	select {
-	case <-d.quitCh:
-		// In case the worker has quit, prevent blocking on the event channel.
-	case d.eventCh <- ev:
-	}
-}
-
-func (d *faultDetector) submit(ctx context.Context) {
-	d.logger.Warn("independently submitting executor commit")
-
-	err := d.submitter.SubmitExecutorCommit(ctx, d.commit)
-	switch err {
-	case nil:
-		d.logger.Info("independently submitted executor commit")
-	default:
-		d.logger.Error("failed to submit executor commit independently",
-			"err", err,
-		)
-	}
-}
-
-func (d *faultDetector) worker(ctx context.Context) {
-	// We should submit the commitment immediately in case when:
-	//
-	// - We see a merge commit that does not have our commitment.
-	// - We see an executor commit.
-	// - RoundTimeout elapses without seeing our commitment.
-	//
-	defer close(d.quitCh)
-
-	// Determine the round timeout and start a local timer.
-	rtDesc, err := d.runtime.RegistryDescriptor(ctx)
-	if err != nil {
-		d.logger.Error("failed to retrieve runtime registry descriptor",
-			"err", err,
-		)
-		return
-	}
-	// Add a small amount to compensate for network latency.
-	timer := time.NewTimer(rtDesc.Executor.RoundTimeout + 1*time.Second)
-
-	// Extract committee ID for easier comparison.
-	openCommit, err := d.commit.Open()
-	if err != nil {
-		// This should NEVER happen.
-		d.logger.Error("bad own commitment",
-			"err", err,
-		)
-		return
-	}
-
-	// TODO: Once we have P2P gossipsub also look at gossiped commitments in addition to consensus.
-
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		case <-timer.C:
-			// Local round timeout expired.
-			d.logger.Warn("local round timeout expired without seeing our commitment")
```

Comment:
what was this file, and why are we removing it as part of this change?

Reply:
Because the point of the fault detector was to independently submit executor commits in case the merge committee is faulty. If there is no merge committee, this is no longer needed.

Reply:
oh
:information_source: now executors submit directly to consensus